pyomo/core/expr/expr_pyomo5.py

```diff
@@ -726,6 +727,36 @@ def dfs_postorder_stack(self, node):
             else:
                 return self.finalize(ans)
 
+def replace_expressions(expr,
+                        substitution_map,
+                        descend_into_named_expressions=True,
+                        remove_named_expressions=False):
+    """
+
+    Parameters
+    ----------
+    expr : Pyomo expression
+       The source expression
+    substitution_map : dict
+       A dictionary mapping object ids in the source to the replacement objects.
+    descend_into_named_expressions : bool
+       True if replacement should go into named expression objects, False to halt at
+       a named expression
+    remove_named_expressions : bool
+       True if the named expressions should be replaced with a standard expression,
+       and False if the named expression should be left in place
+
```

Comment:
The only code I added was to create a wrapper function for the ExpressionReplacementWalker that is already there. I looked into the ExpressionRepalcementWalker, and it will replace the named expression with the little-e expression that it contained.